libcxx/test/std/utilities/optional/optional.relops/relops.compile.pass.cpp

```diff
@@ -0,0 +1,40 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// REQUIRES: std-at-least-c++17
+// <optional>
+
+// Verify that comparison operators of `optional` accept underlying return types convertible both from and to `bool`
+// as required by LWG4370.
+
+#include <optional>
+
+struct Bool {
+  Bool(bool) {};
+  operator bool() const { return true; };
+};
+
+struct S {
+  Bool operator==(S) const { return true; }
+  Bool operator!=(S) const { return true; }
+  Bool operator<=(S) const { return true; }
+  Bool operator<(S) const { return true; }
+  Bool operator>(S) const { return true; }
+  Bool operator>=(S) const { return true; }
+};
+
+constexpr void test() {
```

Comment:
The `constexpr` looks superfluous if we only verify the validness of comparison operator functions.
Also, it makes a program ill-formed, no diagnostic required before C++23 (because the underlying comparisons never result in constant subexpressions, see also [P2448R2](http://wg21.link/p2448R2)).
```suggestion
void test() {
```